### DISCUSSION TITLE
`nx migrate` does not update deps inside nested `package.json`

### DISCUSSION THREAD

**Discussion by nsbarsukov** [nrwl/nx#26693] [Feature Requests]
### Current Behavior

Use project with the following structure:

```
projects
├── lib-1
│   └── package.json
├── lib-2
│   └── package.json
├── demo-application
│   └── package.json  <-- it contains `@taiga-ui/cdk` dependency
└── package.json
```

Root `package.json` contains 
```
"workspaces": [
    "projects/*"
]
```

Run the following command:
```
nx migrate @taiga-ui/cdk@4.0.0-rc.2
```

✅ it will produce `migrations.json` 
❌ However, it will not increment dependencies' versions inside `projects/demo-application/package.json`

### Expected Behavior

Increament deps inside nested `package.json`-s.

### GitHub Repo

https://github.com/taiga-family/maskito

### Steps to Reproduce

1. Clone repository https://github.com/taiga-family/maskito
2. Switch to the branch [`v2`](https://github.com/taiga-family/maskito/tree/v2)
3. Run
    ```
    npm ci
    ```
4. Run 
    ```
    nx migrate @taiga-ui/cdk@4.0.0-rc.2
    ```

### Nx Report

```shell
Node   : 18.19.1
OS     : darwin-arm64
npm    : 10.2.4

nx (global)    : 19.0.3
nx             : 19.0.8
@nx/js         : 19.0.8
@nx/jest       : 19.0.8
@nx/linter     : 19.0.8
@nx/eslint     : 19.0.8
@nx/workspace  : 19.0.8
@nx/angular    : 19.0.8
@nx/cypress    : 19.0.8
@nx/devkit     : 19.0.8
@nx/react      : 19.0.8
@nx/rollup     : 19.0.8
@nrwl/tao      : 19.0.8
@nx/web        : 19.0.8
@nx/webpack    : 19.0.8
typescript     : 4.9.5
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by Coly010** [comment]
Hi, this is functioning as expected. 

Perhaps we could make it an enhancement for the future, but as of right now this is correct
